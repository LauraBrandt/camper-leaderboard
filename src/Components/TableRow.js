import React from 'react';
import PropTypes from 'prop-types';

const TableRow = ({ user, index }) => {
  return(
    <tr className="row">
      <td>{index+1}</td>
      <td>
        <a href={`https://www.freecodecamp.org/${user.username}`} className="user-info">
          <img src={user.img} alt={`${user.username} avatar`}/>
          <p>{user.username}</p>
        </a>
      </td>
      <td>{user.recent}</td>
      <td>{user.alltime}</td>
    </tr>
  );
}

TableRow.propTypes = {
  index: PropTypes.number.isRequired,
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    recent: PropTypes.number.isRequired,
    alltime: PropTypes.number.isRequired,
    img: PropTypes.string
  })
}

export default TableRow;
