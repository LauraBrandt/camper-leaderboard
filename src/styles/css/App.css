.App {
  min-height: 100vh;
  min-width: 100vw;
  background-color: #2A2A2A;
  display: flex;
  flex-direction: column;
  font-family: 'Lato', sans-serif; }

.header {
  background-color: #006400;
  text-align: center;
  color: white;
  padding: 1.5em; }
  .header h1 {
    margin: 0; }

@media (max-width: 450px) {
  .header {
    z-index: 1;
    box-shadow: inset 0 -3px 10px rgba(0, 24, 0, 0.3), inset 0 3px 10px rgba(0, 177, 0, 0.6); } }

.footer {
  background-color: #006400;
  color: #939393;
  text-align: center;
  padding: 0.5em; }
  .footer a {
    text-decoration: none;
    color: #E0E0E0; }
