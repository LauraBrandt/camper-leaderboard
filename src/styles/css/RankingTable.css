.ranking-table-wrapper {
  width: 100%;
  flex: 1 0 auto;
  padding: 3em 0; }

.ranking-table {
  background-color: white;
  width: 90%;
  margin: auto;
  border-radius: 7px;
  border-collapse: collapse;
  border-style: hidden;
  font-size: 1.1rem;
  font-family: 'Source Sans Pro', sans-serif; }

.header-row th {
  color: white;
  padding: 0.8em;
  background-color: #A4755F;
  font-weight: normal;
  font-size: 1.2rem; }
  .header-row th:first-child {
    border-top-left-radius: 7px; }
  .header-row th:last-child {
    border-top-right-radius: 7px; }

.row td {
  border: 2px solid rgba(164, 117, 95, 0.4);
  padding: 0.3em;
  text-align: center; }

.row:nth-child(even) {
  background-color: rgba(224, 224, 224, 0.5); }

.row:last-child td:first-child {
  border-bottom-left-radius: 7px; }

.row:last-child td:last-child {
  border-bottom-right-radius: 7px; }

.row:hover {
  background-color: rgba(164, 117, 95, 0.3); }

.user-info {
  text-decoration: none;
  color: #3949AB;
  display: flex;
  align-items: center; }
  .user-info img {
    width: 45px;
    height: 45px;
    border: 1px solid rgba(147, 147, 147, 0.4);
    border-radius: 3px;
    margin: 0.3em 1.5em 0.3em 1em; }
  .user-info p {
    margin: 0.5em 0; }

@media (max-width: 450px) {
  .ranking-table-wrapper {
    padding: 0; }
  .ranking-table {
    width: 100%;
    font-size: 1rem;
    border-radius: 0; }
    .ranking-table .header-row th {
      font-size: 1rem;
      border-radius: 0; }
    .ranking-table .row td {
      border-radius: 0; }
  .user-info {
    flex-direction: column; }
    .user-info img {
      margin: 0.5em 0 0 0; } }
