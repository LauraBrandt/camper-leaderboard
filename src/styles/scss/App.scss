@import '_variables';

.App {
  min-height: 100vh;
  min-width: 100vw;
  background-color: $fcc-black;
  display: flex;
  flex-direction: column;
  font-family: 'Lato', sans-serif;
}

.header {
  background-color: $fcc-green;
  text-align: center;
  color: white;
  padding: 1.5em;
  h1 {
    margin: 0;
  }
}

@media (max-width: 450px) {
  .header {
    z-index: 1;
    box-shadow: inset 0 -3px 10px rgba(darken($fcc-green, 15%), 0.3),
                inset 0 3px 10px rgba(lighten($fcc-green, 15%), 0.6);
  }
}

.footer {
  background-color: $fcc-green;
  color: $fcc-dark-grey;
  text-align: center;
  padding: 0.5em;
  a {
    text-decoration: none;
    color: $fcc-grey;
  }
}
