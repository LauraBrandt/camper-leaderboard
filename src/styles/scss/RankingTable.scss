@import '_variables';

.ranking-table-wrapper {
  width: 100%;
  flex: 1 0 auto;
  padding: 3em 0;
}

.ranking-table {
  background-color: white;
  width: 90%;
  margin: auto;
  border-radius: $table-border-radius;
  border-collapse: collapse;
  border-style: hidden;
  font-size: 1.1rem;
  font-family: 'Source Sans Pro', sans-serif;
}


.header-row {
  th {
    color: white;
    padding: 0.8em;
    background-color: $fcc-brown;
    font-weight: normal;
    font-size: 1.2rem;
    &:first-child {
      border-top-left-radius: $table-border-radius;
    }
    &:last-child {
      border-top-right-radius: $table-border-radius;
    }
  }
}

.row {
  td {
    border: 2px solid rgba($fcc-brown, 0.4);
    padding: 0.3em;
    text-align: center;
  }
  &:nth-child(even) {
    background-color: rgba($fcc-grey, 0.5);
  }
  &:last-child {
    td:first-child {
      border-bottom-left-radius: $table-border-radius;
    }
    td:last-child {
      border-bottom-right-radius: $table-border-radius;
    }
  }
  &:hover {
    background-color: rgba($fcc-brown, 0.3);
  }
}

.user-info {
  text-decoration: none;
  color: $fcc-blue;
  display: flex;
  align-items: center;
  img {
    width: $image-size;
    height: $image-size;
    border: 1px solid rgba($fcc-dark-grey, 0.4);
    border-radius: 3px;
    margin: 0.3em 1.5em 0.3em 1em;
  }
  p {
    margin: 0.5em 0;
  }
}

@media (max-width: 450px) {
  .ranking-table-wrapper {
    padding: 0;
  }
  .ranking-table {
    width: 100%;
    font-size: 1rem;
    border-radius: 0;
    .header-row th {
      font-size: 1rem;
      border-radius: 0;
    }
    .row td {
      border-radius: 0;
    }
  }
  .user-info {
    flex-direction: column;
    img {
      margin: 0.5em 0 0 0;
    }
  }
}