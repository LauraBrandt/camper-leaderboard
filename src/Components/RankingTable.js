import React from 'react';
import TableRow from './TableRow';
import users from '../data';

class RankingTable extends React.Component {
  constructor() {
    super();
    this.state = {
      users: []
    }
    this.renderRows = this.renderRows.bind(this);
  }

  componentDidMount() {
    this.setState({ users });
  }

  renderRows() {
    return this.state.users.map((user, i) => 
      <TableRow user={user} index={i} key={user.username} />
    )
  }

  render() {
    return (
      <div className="ranking-table-wrapper">
        <table className="ranking-table">
          <thead>
            <tr className="row header-row">
              <th>#</th>
              <th>Camper</th> 
              <th>Points in last 30 days</th>
              <th>All time points</th>
            </tr>
          </thead>
          <tbody>
            {this.state.users && this.renderRows()}
          </tbody>
        </table>
      </div>
    );
  }
}

export default RankingTable;
