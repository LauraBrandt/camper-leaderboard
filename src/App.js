import React from 'react';
import RankingTable from './Components/RankingTable';

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <header className="header">
          <h1>FCC Camper Leaderboard</h1>
        </header>
        <RankingTable />
        <footer className="footer">Designed and coded by <a href="https://github.com/LauraBrandt">Laura Brandt</a></footer>
      </div>
    );
  }
}

export default App;
