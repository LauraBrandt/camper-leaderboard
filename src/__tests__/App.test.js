import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../App';
import RankingTable from '../Components/RankingTable';
import TableRow from '../Components/TableRow';

describe('App', () => {
  it('renders and matches snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('content', () => {
    let component;
    beforeEach(() => {
      component = shallow(<App />);
    });

    it('has a header', () => {
      expect(component.find('header')).toHaveLength(1);
    });

    it('has a footer', () => {
      expect(component.find('footer')).toHaveLength(1);
    });

    it('has 1 RankingTable component', () => {
      expect(component.find(RankingTable)).toHaveLength(1);
    });
  });
});
