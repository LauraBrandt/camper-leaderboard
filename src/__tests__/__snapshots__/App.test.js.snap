// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1>
      FCC Camper Leaderboard
    </h1>
  </header>
  <div
    className="ranking-table-wrapper"
  >
    <table
      className="ranking-table"
    >
      <thead>
        <tr
          className="row header-row"
        >
          <th>
            #
          </th>
          <th>
            Camper
          </th>
          <th>
            Points in last 30 days
          </th>
          <th>
            All time points
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="row"
        >
          <td>
            1
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/iulius83"
            >
              <img
                alt="iulius83 avatar"
                src="https://avatars2.githubusercontent.com/u/36899653?v=4"
              />
              <p>
                iulius83
              </p>
            </a>
          </td>
          <td>
            45
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            2
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/Faithyoussef"
            >
              <img
                alt="Faithyoussef avatar"
                src="https://avatars0.githubusercontent.com/u/38753427?v=4"
              />
              <p>
                Faithyoussef
              </p>
            </a>
          </td>
          <td>
            45
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            3
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/RecursiveRach"
            >
              <img
                alt="RecursiveRach avatar"
                src="https://avatars3.githubusercontent.com/u/34367085?v=4"
              />
              <p>
                RecursiveRach
              </p>
            </a>
          </td>
          <td>
            44
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            4
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/hummeldon"
            >
              <img
                alt="hummeldon avatar"
                src="https://avatars2.githubusercontent.com/u/38077994?v=4"
              />
              <p>
                hummeldon
              </p>
            </a>
          </td>
          <td>
            44
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            5
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/kpone1"
            >
              <img
                alt="kpone1 avatar"
                src="https://avatars1.githubusercontent.com/u/38334216?v=4"
              />
              <p>
                kpone1
              </p>
            </a>
          </td>
          <td>
            44
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            6
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/karthikd69"
            >
              <img
                alt="karthikd69 avatar"
                src="https://avatars3.githubusercontent.com/u/30834118?v=4"
              />
              <p>
                karthikd69
              </p>
            </a>
          </td>
          <td>
            44
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            7
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/Sambender9"
            >
              <img
                alt="Sambender9 avatar"
                src="https://avatars.githubusercontent.com/u/7374344?v=3"
              />
              <p>
                Sambender9
              </p>
            </a>
          </td>
          <td>
            43
          </td>
          <td>
            44
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            8
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/hugomartinez"
            >
              <img
                alt="hugomartinez avatar"
                src="https://avatars2.githubusercontent.com/u/3845667?v=3"
              />
              <p>
                hugomartinez
              </p>
            </a>
          </td>
          <td>
            42
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            9
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/KibaAmor"
            >
              <img
                alt="KibaAmor avatar"
                src="https://avatars0.githubusercontent.com/u/5093911?v=3"
              />
              <p>
                KibaAmor
              </p>
            </a>
          </td>
          <td>
            42
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            10
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/zhangliting"
            >
              <img
                alt="zhangliting avatar"
                src="https://avatars1.githubusercontent.com/u/16881281?v=3"
              />
              <p>
                zhangliting
              </p>
            </a>
          </td>
          <td>
            42
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            11
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/blastit"
            >
              <img
                alt="blastit avatar"
                src="https://avatars2.githubusercontent.com/u/23505048?v=3"
              />
              <p>
                blastit
              </p>
            </a>
          </td>
          <td>
            41
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            12
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/omerharshit"
            >
              <img
                alt="omerharshit avatar"
                src="https://avatars3.githubusercontent.com/u/20610736?v=4"
              />
              <p>
                omerharshit
              </p>
            </a>
          </td>
          <td>
            41
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            13
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/valigabi"
            >
              <img
                alt="valigabi avatar"
                src="https://avatars1.githubusercontent.com/u/33953042?v=4"
              />
              <p>
                valigabi
              </p>
            </a>
          </td>
          <td>
            41
          </td>
          <td>
            42
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            14
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/GorkemOkur"
            >
              <img
                alt="GorkemOkur avatar"
                src="https://avatars3.githubusercontent.com/u/16239094?v=4"
              />
              <p>
                GorkemOkur
              </p>
            </a>
          </td>
          <td>
            40
          </td>
          <td>
            41
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            15
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/v01dMage"
            >
              <img
                alt="v01dMage avatar"
                src="https://avatars1.githubusercontent.com/u/8935634?v=4"
              />
              <p>
                v01dMage
              </p>
            </a>
          </td>
          <td>
            39
          </td>
          <td>
            44
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            16
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/Ampersandme"
            >
              <img
                alt="Ampersandme avatar"
                src="https://avatars1.githubusercontent.com/u/32043406?v=4"
              />
              <p>
                Ampersandme
              </p>
            </a>
          </td>
          <td>
            36
          </td>
          <td>
            37
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            17
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/AloisJanicek"
            >
              <img
                alt="AloisJanicek avatar"
                src="https://avatars3.githubusercontent.com/u/804158?v=4"
              />
              <p>
                AloisJanicek
              </p>
            </a>
          </td>
          <td>
            35
          </td>
          <td>
            45
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            18
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/ijhar8"
            >
              <img
                alt="ijhar8 avatar"
                src="https://avatars0.githubusercontent.com/u/29692615?v=4"
              />
              <p>
                ijhar8
              </p>
            </a>
          </td>
          <td>
            34
          </td>
          <td>
            35
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            19
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/cmccormack"
            >
              <img
                alt="cmccormack avatar"
                src="https://avatars1.githubusercontent.com/u/3016919?v=4"
              />
              <p>
                cmccormack
              </p>
            </a>
          </td>
          <td>
            34
          </td>
          <td>
            1135
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            20
          </td>
          <td>
            <a
              className="user-info"
              href="https://www.freecodecamp.org/AzureXX"
            >
              <img
                alt="AzureXX avatar"
                src="https://avatars2.githubusercontent.com/u/32188281?v=4"
              />
              <p>
                AzureXX
              </p>
            </a>
          </td>
          <td>
            33
          </td>
          <td>
            45
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <footer
    className="footer"
  >
    Designed and coded by 
    <a
      href="https://github.com/LauraBrandt"
    >
      Laura Brandt
    </a>
  </footer>
</div>
`;
